sauce/features/budget/resize-inspector/index.js

```diff
@@ -0,0 +1,219 @@
+const BUTTONDISABLED = 'button-disabled';
+
+export class ResizeInspector extends Feature {
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  shouldInvoke() {
+    return toolkitHelper.getCurrentRouteName().indexOf('budget') !== -1 &&
+           this.settings.enabled;
+  }
+
+  invoke() {
+    let width = toolkitHelper.getToolkitStorageKey('inspector-width', 'number');
+    if (typeof width === 'undefined' || width === null) {
+      width = 0;
+    }
+
+    this.setProperties(width);
+    this.addResizeButton();
+  }
+
+  addResizeButton() {
+    if (!$('#toolkitResizeInspector').length) {
+      let buttonText = ynabToolKit.l10nData && ynabToolKit.l10nData['toolkit.InspectorWidth'] || 'Inspector Width';
+      $('<button>', { id: 'toolkitResizeInspector', class: 'ember-view button', style: 'float: right' })
+        .append(buttonText + ' ')
+        .append($('<i>', { class: 'ember-view flaticon stroke down' }))
+        .click(() => {
+          this.showResizeModal();
+        })
+        .insertAfter('.undo-redo-container');
+    }
+  }
+
+  showResizeModal() {
+    let _this = this;
+    let btnLeft = $('.budget-toolbar').outerWidth() + $('#toolkitResizeInspector').outerWidth() + 24;
+    let btnTop = $('.budget-toolbar').outerHeight() + $('.budget-header-flexbox').outerHeight() + 8;
+    let $modal = $('<div id=toolkitInspectorODiv class="ember-view">' +
+                    '<div id="toolkitInspectorModal" class="ynab-u modal-popup modal-resize-inspector ember-view modal-overlay active">' +
+                     '<div id=toolkitInspectorIDiv class="modal" style="left: ' + btnLeft + 'px; top: ' + btnTop + 'px;">' +
+                      '<ul class="modal-list">' +
+                        '<li><button id="toolkitInspector00Btn" class="button-list"><i id=toolkitInspector00Img class="ember-view flaticon stroke checkmark-1"/>Default</button></li>' +
+                        '<li><button id="toolkitInspector25Btn" class="button-list"><i id=toolkitInspector25Img class="ember-view flaticon stroke checkmark-1"/>25%</button></li>' +
+                        '<li><button id="toolkitInspector20Btn" class="button-list"><i id=toolkitInspector20Img class="ember-view flaticon stroke checkmark-1"/>20%</button></li>' +
+                        '<li><button id="toolkitInspector15Btn" class="button-list"><i id=toolkitInspector15Img class="ember-view flaticon stroke checkmark-1"/>15%</button></li>' +
+                      '</ul>' +
+                      '<div class="modal-arrow" style="position:absolute;width: 0;height: 0;bottom: 100%;left: 37px;border: solid transparent;border-color: transparent;border-width: 15px;border-bottom-color: #fff"></div>' +
+                     '</div>' +
+                    '</div>' +
+                   '</div>');
+
+    let $00img = $modal.find('#toolkitInspector00Img');
+    let $00btn = $modal.find('#toolkitInspector00Btn');
+    $00btn.click(() => {
+      _this.setProperties(0);
+    });
```

Comment:
All of these things can probably be better handled by a `data` attribute on the DOM items themselves. So each `<li>` could have a `data-inspector-size="0"` or `data-inspector-size="1"` etc. Then you can assign the same `click()` handler to all of them that goes `_this.setProperties($(this).data('inspector-size'))`, and you can also do a selector across all buttons to set them not selected, then use a selector like `$("li[data-insepctor-size='2'])` to find the one that's selected and set it that way.

Just trying to avoid having specific variables for all the things if we can. Happy to help if I'm not making sense as well.

Reply:
Nah, makes perfect sense and a much better approach.